google/cloud/spanner/internal/connection_impl_test.cc

```diff
@@ -706,88 +728,132 @@ TEST(ConnectionImplTest, ExecuteQueryImplicitBeginTransaction) {
   EXPECT_THAT(txn, HasSessionAndTransactionId("test-session-name", "00FEDCBA"));
 }
 
+/**
+ * @test Verify that the protos sent by the 9 Connection::* methods affected
+ * by QueryOptions contain the expected fields.
+ */
 TEST(ConnectionImplTest, QueryOptions) {
-  auto constexpr kQueryOptionsProp =
-      &spanner_proto::ExecuteSqlRequest::query_options;
-  auto constexpr kRequestOptionsProp =
-      &spanner_proto::ExecuteSqlRequest::request_options;
-
-  // Helper functions to build QueryOptions/RequestOptions protos inline.
-  auto const make_qo_proto = [](absl::optional<std::string> version,
-                                absl::optional<std::string> stats) {
-    spanner_proto::ExecuteSqlRequest::QueryOptions proto;
-    if (version) proto.set_optimizer_version(*version);
-    if (stats) proto.set_optimizer_statistics_package(*stats);
-    return proto;
-  };
-  auto const make_ro_proto =
-      [](absl::optional<spanner_proto::RequestOptions::Priority> priority) {
-        spanner_proto::RequestOptions proto;
-        if (priority) proto.set_priority(*priority);
-        return proto;
-      };
-
   struct {
-    // Given this for SqlParams::query_options ...
+    // Given these QueryOptions ...
     spanner::QueryOptions options;
-
-    // ... expect these for ExecuteSqlRequest query_options/request_options.
+    // ... expect these protos.
     spanner_proto::ExecuteSqlRequest::QueryOptions qo_proto;
     spanner_proto::RequestOptions ro_proto;
   } test_cases[] = {
```

Comment:
At some point maybe you would need to split this test cases into separate function, and use `TEST_P()` with the parameter being the values in this array.  Just a thought for the future, right now it is probably six of one, half a dozen of the other.

Reply:
Acknowledged.